pkg/util/limiter/query_limiter.go

```diff
@@ -19,18 +19,23 @@ import (
 
 type queryLimiterCtxKey struct{}
 
+const cardinalityStrategy = "Consider reducing the time range and/or cardinality of the query. To reduce the query cardinality, you can add more label matchers to the query, or increase the number of query shards"
```

Comment:
I'm wondering if the concept of "cardinality" is something we expect users to understand, or if there's another term that would be clearer for them.

We should probably also include a hint on how to increase the number of query shards, and make it clear this is a parallel action to reducing cardinality.

Perhaps rephrasing this message to something like this would be clearer:

```suggestion
const cardinalityStrategy = "Consider reducing the time range and/or number of series selected by the query, and/or increase the number of query shards. To reduce the number of series selected, add more label matchers to the query. To increase the number of query shards, increase the value of -<sharding option name here>"
```

Reply:
That does look clearer I guess. Though, instead of "increase the value of -<sharding option name here>", should we instead say to contact the service administrator, since it's not obvious how to increase the number of query shards?

Reply:
I'll commit the suggestion, and try to edit it a bit based on feedback. Watch this space.